Assets/Scripts/GraphService.cs

```diff
@@ -180,8 +181,26 @@ void FixEdgeGroup(IList<Edge> edges, Node nodeFrom, Node nodeTo)
         lineRenderer.positionCount = linePositionsCount;
         lineRenderer.SetPositions(linePositions[i].ToArray());
 
-        var meshCollider = edges[i].gameObject.GetComponent<MeshCollider>();
-        lineRenderer.BakeMesh(meshCollider.sharedMesh, true);
+        UpdateEdgeColliders(edges[i], linePositions[i]);
+      }
+    }
+
+    void UpdateEdgeColliders(Edge edge, IList<Vector3> linePositions)
+    {
+      var edgeColliderTransforms = edge.gameObject.GetComponentsInChildren<Transform>()
+        .Where(t => t.gameObject.name == Constants.ColliderGameObjectName).ToList();
+
+      for (int i = 0; i < edgeColliderTransforms.Count; i++)
```

Comment:
I don;t like such if branches in for loops. Better would be to loop to count - 1 and execute the first branch instructions, and then Destroy edgeColliderTransforms.Last.gameObject outside the loop.